### PR TITLE
feat: Add 'styleTree' function

### PR DESCRIPTION
**Pull request by markdalgleish**

### CODE REVIEW

File: packages/treat/tests/__snapshots__/styling.test.ts.snap [hunk 1]
```diff
@@ -159,6 +159,24 @@ Object {
...
+  "border-image-width": "initial",
+  "border-left-color": "pink",
+  "border-left-style": "solid",
+  "border-left-width": "6px",
+  "border-right-color": "pink",
+  "border-right-style": "solid",
+  "border-right-width": "6px",
+  "border-top-color": "pink",
+  "border-top-style": "solid",
+  "border-top-width": "6px",
```
Comment (mattsjones) [inline packages/treat/tests/__snapshots__/styling.test.ts.snap]: Shorthand properties tend to be pretty noisy in the snapshots so I've avoided until now. Thoughts?

Comment (markdalgleish) [reply packages/treat/tests/__snapshots__/styling.test.ts.snap]: Yep, let's clean this up 👍

File: packages/treat/src/builder.ts [hunk 2]
```diff
@@ -274,3 +275,73 @@ export function globalStyle(
...
+  } catch (err) {
+    throw new Error(
+      "Return values from 'styleTree' functions must only contain objects, arrays and primitive types.",
+    );
+  }
+
+  const [referenceTree, ...restTrees] = themedTrees;
+  restTrees.forEach(tree => {
+    if (!isEqual(tree, referenceTree)) {
+      throw new Error(dedent`
```
[the remark this answers is not in the document]
Comment (markdalgleish) [reply packages/treat/src/builder.ts]: I dunno, I don't mind them being defined where needed. Not sure there's much to gain by pulling them out.

[the remark this answers is not in the document]
Comment (mattsjones) [reply packages/treat/src/builder.ts]: Mmm I understand. Kinda torn on it. Can always refactor later.

**Comment by markdalgleish** [comment]
Realised we missed local debug name support. Just added that now.

**Pull request merged**